src/Collector.cs

```diff
@@ -93,17 +93,11 @@ private void WorkerOnDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
           try
             {
               socket = Connect();
-              if(!Authenticate(socket))
-                {
-                  socket.Close();
-                  // Throttle calls to authenticate so that you do not flood the buffer if collector is down/slow for a few seconds.
-                  // Ultimately should share logic with the backoff stuff below, but did not want to just throw here.
-                  Thread.Sleep(5000);
-                  continue;
-                }
+              Authenticate(socket);
               SendQueuedMessages(socket);
-              failures = 0;
+              socket.Shutdown(SocketShutdown.Both);
               socket.Close();
+              failures = 0;
```

Comment:
The explanation makes sense, but I think the only way this line is executed is when the client is shutting down as noted above. I think a successful auth should be where the failures get reset to 0, so that a connection failure there-after would cause instant retry.